src/megatron/bridge/data/loaders.py

```diff
@@ -137,7 +137,9 @@ def get_train_valid_test_num_samples(cfg: ConfigContainer) -> tuple[int, int, in
 
 
 def build_train_valid_test_datasets(
-    cfg: ConfigContainer, build_train_valid_test_datasets_provider: Callable
+    cfg: ConfigContainer,
+    build_train_valid_test_datasets_provider: Callable,
+    train_state: TrainState,
```

Comment:
what's the reason that we need to pass in train_state? we don't want to pass large object in unless necessary.